adapters/httpapi_curl.c

```diff
@@ -904,6 +904,26 @@ HTTPAPI_RESULT HTTPAPI_SetOption(HTTP_HANDLE handle, const char* optionName, con
                 }
             }
         }
+        else if (strcmp(OPTION_CURL_INTERFACE, optionName) == 0)
+        {
+            const char *interfaceName = (const char*)value;
+            if (interfaceName != NULL && interfaceName[0] != '\0')
+            {
+                if (curl_easy_setopt(httpHandleData->curl, CURLOPT_INTERFACE, interfaceName) != CURLE_OK)
+                {
+                    LogError("unable to curl_easy_setopt for CURLOPT_INTERFACE");
+                    result = HTTPAPI_ERROR;
+                }
+                else
+                {
+                    result = HTTPAPI_OK;
+                }
+            }
+            else
+            {
+                result = HTTPAPI_OK;
```

Comment:
We should return `HTTPAPI_ERROR` here along with a LogError().  

For some more random context here -- it's true of the code above around trusted certs was returning OK if values were not NULL I know, but it's tricky because in one of cases the x509certificate is set but then the x509privateKey is checked against NULL and return OK if it's NULL.  This does this because both values need to be set.  (We really should've made a single option for cert/private key in a struct{} so one OPTION set this, instead of the tearing and needing multiple SetOptions.  But that's a digression).

```c
            httpHandleData->x509certificate = value;
            if (httpHandleData->x509privatekey != NULL)
```

Reply:
done